File: v2/pkg/protocols/network/protocols/mssql/mssql.go
package mssql

import (
	"database/sql"
	"fmt"

	_ "github.com/microsoft/go-mssqldb"
	"github.com/pkg/errors"
)

// ConnectWithCredentials connects to a server with credentials
func ConnectWithCredentials(host, username, password string, port, timeout int) (bool, error) {
	url := fmt.Sprintf("server=%s;port=%d;user id=%s;password=%s;database=master;connection timeout=%d", host, port, username, password, timeout)

	db, err := sql.Open("mssql", url)
	if err != nil {
		return false, errors.Wrap(err, "could not connect to mssql")
	}
	defer func() {
		_ = db.Close()
	}()

	if err = db.Ping(); err != nil {
		return false, errors.Wrap(err, "could not ping mssql")
	}
	return true, nil
}

File: pkg/js/devtools/tsdocs/cmd/tsdocs/src/mssql.ts

/**
 * MSSQLClient Class
 */
export class MSSQLClient {
    

    /**
    * Connect connects to MS SQL database using given credentials.
    * If connection is successful, it returns true.
    * If connection is unsuccessful, it returns false and error.
    * The connection is closed after the function returns.
    * @throws {Error} - if the operation fails
    */
    public Connect(host: string, port: number, username: string): boolean | null {
        return null;
    }
    

    /**
    * ConnectWithDB connects to MS SQL database using given credentials and database name.
    * If connection is successful, it returns true.
    * If connection is unsuccessful, it returns false and error.
    * The connection is closed after the function returns.
    * @throws {Error} - if the operation fails
    */
    public ConnectWithDB(host: string, port: number, username: string): boolean | null {
        return null;
    }
    

    /**
    * IsMssql checks if the given host is running MS SQL database.
    * If the host is running MS SQL database, it returns true.
    * If the host is not running MS SQL database, it returns false.
    * @throws {Error} - if the operation fails
    */
    public IsMssql(host: string, port: number): boolean | null {
        return null;
    }
    

}

